R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -33,54 +34,191 @@ module_socioeconomics_L100.Population_downscale_ctry_DISABLED <- function(comman
+      complete(year, nesting(iso)) %>%  # Completes tibble to include all years for all iso (creates missing values)
+      filter(!is.na(iso)) %>%  # iso NA values created with complete on year and iso table
+      arrange(iso, year) %>%  # For easier readability when checking results
+      group_by(iso) %>%
+      mutate(pop = approx_fun(year, pop)) %>%  # Interpolate -- note that there will still be missing values for countries that do not have end values (1500, 1600, or 1700)
+      mutate(pop = if_else(is.na(pop) & year == 1800, pop[year == 1820], pop)) %>%  # Replace missing 1800 values with 1820 for countries that begin in 1820 (note that Panama has a value in 1820 = 0, not NA)
+      filter(year %in% c(socioeconomics.MADDISON_HISTORICAL_YEARS, socioeconomics.AGG_BASE_YEAR)) # Keep only needed years
+
+    # Fourth, assign population values to countries with missing values in historical years
+    # Extract total world population for historic years
+    pop_global <- filter(hist_interp, iso == "world_total") %>%
+      ungroup() %>%
+      select(-iso)
+    # Sum all countries' populations in each period
+    pop_notmissing <- filter(hist_interp, iso != "world_total") %>%
+      group_by(year) %>%
+      mutate(pop = if_else(is.na(pop), 0, pop)) %>%
+      summarize(pop = sum(pop))
+    # Subtract reported countries from total global - these are the total population values in each period that will be allocated to countries with missing values
+    pop_missing <- left_join(pop_global, pop_notmissing, by = "year") %>%
+      mutate(pop_allocate = pop.x - pop.y) %>%
+      select(year, pop_allocate) %>%
+      mutate(year = paste("X", year, sep = "")) %>%
+      spread(year, pop_allocate)
+
+    # Fifth, estimate historic population values for countries with missing values and calculate ratios relative to 1950
+    maddison_hist_ratio <- filter(hist_interp, iso != "world_total") %>%
+      ungroup() %>%
+      mutate(year = paste("X", year, sep = "")) %>%
+      spread(year, pop) %>%
+      # Fill in missing values for period t using ratio of country to total population in period t+1 times the missing share in period t.
+      # if(is.na(X1900), sum(X1950))
+      mutate(X1900 = if_else(is.na(X1900), ((X1950 / sum(X1950[is.na(X1900)])) * pop_missing$X1900), X1900)) %>%
+      mutate(X1850 = if_else(is.na(X1850), ((X1900 / sum(X1900[is.na(X1850)])) * pop_missing$X1850), X1850)) %>%
+      mutate(X1800 = if_else(is.na(X1800), ((X1850 / sum(X1850[is.na(X1800)])) * pop_missing$X1800), X1800)) %>%
+      mutate(X1750 = if_else(is.na(X1750), ((X1800 / sum(X1800[is.na(X1750)])) * pop_missing$X1750), X1750)) %>%
+      mutate(X1700 = if_else(is.na(X1700), ((X1750 / sum(X1750[is.na(X1700)])) * pop_missing$X1700), X1700)) %>%
+      # Population ratios for historic years compared to 1950 for all countries (will be used to scale UN populations from 1950)
```

Comment:
ok. i think this one has been held up for too long. I'd prefer not to hard-code the years, but that's just a stylistic issue. 
as far as i'm concerned this reasonably replicates the old data system, and that's all I need to approve this.